pkg/signature/kms/azure/client.go

```diff
@@ -304,21 +304,47 @@ func (a *azureVaultClient) createKey(ctx context.Context) (crypto.PublicKey, err
 	return a.public(ctx)
 }
 
-func getKeyVaultSignatureAlgo(algo crypto.Hash) (keyvault.JSONWebKeySignatureAlgorithm, error) {
-	switch algo {
+func (a *azureVaultClient) getKeyVaultHashFunc() (crypto.Hash, error) {
+	publicKey, err := a.public(context.TODO())
+	if err != nil {
+		return 0, fmt.Errorf("failed to get public key: %w", err)
+	}
+	switch eckey := publicKey.(type) {
+	case *ecdsa.PublicKey:
+		switch eckey.Params().Name {
```

Comment:
nit, I'd compare against `eckey.Curve`, each case statement checking if it's `elliptic.P256()` or 384 or 521.